src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/SubscribeToCertificateUpdatesTest.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    @AfterEach
+    void afterEach() {
+        LogConfig.getRootLogConfig().reset();
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_two_brokers_WHEN_subscribed_to_certificate_updates_THEN_both_receive_updates_on_stream()
+            throws Exception {
+        startNucleusWithConfig("cda.yaml");
+        Map<String, String> broker1Certs = new HashMap<>();
+        Map<String, String> broker2Certs = new HashMap<>();
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerSubscribingToCertUpdates")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+            SubscribeToCertificateUpdatesRequest request = getSampleSubsRequest();
+            Pair<CompletableFuture<Void>, Consumer<CertificateUpdate>> cb = asyncAssertOnConsumer((m) -> {
+                broker1Certs.put("certificate", m.getCertificate());
+                broker1Certs.put("ca-certificate", m.getCaCertificates().get(0));
+            });
+
+            subscribeToCertUpdates(ipcClient, request, cb.getRight());
+            cb.getLeft().get(10, TimeUnit.SECONDS);
+        }
+
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "Broker2SubscribingToCertUpdates")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+            SubscribeToCertificateUpdatesRequest request = getSampleSubsRequest();
+            Pair<CompletableFuture<Void>, Consumer<CertificateUpdate>> cb = asyncAssertOnConsumer((m) -> {
+                broker2Certs.put("certificate", m.getCertificate());
+                broker2Certs.put("ca-certificate", m.getCaCertificates().get(0));
+            });
+
+            subscribeToCertUpdates(ipcClient, request, cb.getRight());
+            cb.getLeft().get(10, TimeUnit.SECONDS);
+            assertThat(broker1Certs.get("ca-certificate"),is(broker2Certs.get("ca-certificate")));
+            assertThat(broker1Certs.get("certificate"),is(not(broker2Certs.get("certificate"))));
+        }
+    }
+
+    @Test
+    void GIVEN_broker_WHEN_subscribed_to_certificate_updates_THEN_certificate_is_generated() throws Exception {
+        startNucleusWithConfig("cda.yaml");
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerSubscribingToCertUpdates")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+            SubscribeToCertificateUpdatesRequest request = getSampleSubsRequest();
+            Pair<CompletableFuture<Void>, Consumer<CertificateUpdate>> cb = asyncAssertOnConsumer((m) -> {
+                assertTrue(m.getCertificate().startsWith("-----BEGIN CERTIFICATE-----"));
+                assertTrue(m.getCaCertificates().get(0).startsWith("-----BEGIN CERTIFICATE-----"));
```

Comment:
you undid the good change. use hamcrest for assertions. There is a startsWith assertion matcher